addons/website_sale/static/src/js/website_sale_utils.js

```diff
@@ -32,7 +36,67 @@ function animateClone($cart, $elem, offsetTop, offsetLeft) {
     }
 }
 
+/**
+ * Allows to propose the existing products from the database.
+ *
+ * @param {ServicesMixin|Widget} self - an element capable to trigger an RPC
+ * @param {jQuery} $input
+ */
+function autocompleteWithProducts(self, $input, data) {
+    $input.autocomplete({
+        source: function (request, response) {
+            return self._rpc({
+                route: '/website_sale/get_products_info',
+                params: {
+                    term: request.term,
+                    limit: 7,
+                    product_options: {
+                        'display_description': data.displayDescription,
+                        'display_price': data.displayPrice,
+                        'display_image': data.displayImage,
+                        'order_by': data.orderBy,
+                    }
+                },
+            }).then((function (resources) {
+                this.products_options = resources.products_options;
+                this.products_info = resources.products_info;
+                var orderBy = this.products_options.order_by;
+                if (orderBy === 'name asc') {
+                    this.products_info.sort();
+                } else if (orderBy === 'name desc') {
+                    this.products_info.reverse();
+                } else if (orderBy === 'list_price asc') {
+                    orderByPrice('low_to_high');
+                } else {
+                    orderByPrice('high_to_low');
+                }
+                response(this.products_info);
```

Comment:
You should not have to sort the elements in JS, what is the `order_by` attribute you sent to the server for ?